### PR TITLE
#160845516 enable users to report articles

### PR DESCRIPTION
**Pull request by collinewait**
#### What does this PR do?
- It enables users to report articles

#### Description of Task to be completed?
- Enable users to report articles that violate terms of the agreement or are plagiarised to the authors haven admin.
 
#### How should this be manually tested?
- Run `npm install `
- Run `npm start`

#### What are the relevant pivotal tracker stories?
[#160845516](https://www.pivotaltracker.com/story/show/160845516)

image

### CODE REVIEW

File: src/components/articles/reportArticleForm.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import * as PropTypes from 'prop-types';
```
Comment (reiosantos) [inline src/components/articles/reportArticleForm.js]: There is no need of `*` in this import. you can instead do `import PropTypes from 'prop-types;'`

File: src/components/articles/reportArticleForm.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import styles from '../../styles/articleStyles/reportArticleForm.scss';
+
+const InputFields = (props) => (
+	<div>
+		<input
+			type='radio'
+			id={props.id}
```
Comment (reiosantos) [inline src/components/articles/reportArticleForm.js]: Use destructuring assignment here. @collinewait

File: src/components/articles/reportArticleForm.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		/>
+		<label htmlFor={props.id} className='black-text'>{props.value}</label>
+	</div>
+);
+
+const ReportArticleForm = (props) =>
+
+	(
+		<form onSubmit={props.handleSubmit} className='report_form'>
+			<InputFields id='immoral' value='It is sexually inappropriate' issue={props.issue} handleChange={props.handleChange} />
```
Comment (reiosantos) [inline src/components/articles/reportArticleForm.js]: The same here `destructuring`.

File: src/components/articles/viewArticleFooter.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ import * as PropTypes from 'prop-types';
 import styles from '../../styles/articleStyles/viewArticle.scss';
 import {USERNAME_KEY} from '../../constants/index';
```
Comment (reiosantos) [inline src/components/articles/viewArticleFooter.js]: We do not have to import from `index`. You can remove the `/index`

File: src/tests/actions/articles/reportArticleActions.test.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import reduxThunk from 'redux-thunk';
+import configurestore from 'redux-mock-store';
+import * as moxios from 'moxios';
```
Comment (reiosantos) [inline src/tests/actions/articles/reportArticleActions.test.js]: Please remove the `*` from this import, and from all places where it doesn't need to be used.

**Review by reiosantos — CHANGES_REQUESTED** [review]

File: src/components/articles/reportArticleForm.js (hunk 1 above)
Comment (collinewait) [reply src/components/articles/reportArticleForm.js]: It has been worked on

File: src/components/articles/reportArticleForm.js (hunk 2 above)
Comment (collinewait) [reply src/components/articles/reportArticleForm.js]: resolved

File: src/components/articles/reportArticleForm.js (hunk 3 above)
Comment (collinewait) [reply src/components/articles/reportArticleForm.js]: resolved

File: src/components/articles/viewArticleFooter.js (hunk 4 above)
Comment (collinewait) [reply src/components/articles/viewArticleFooter.js]: resolved

File: src/tests/actions/articles/reportArticleActions.test.js (hunk 5 above)
Comment (collinewait) [reply src/tests/actions/articles/reportArticleActions.test.js]: resolved

File: public/index.html [hunk 6]
```diff
@@ -23,6 +23,12 @@
     <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
     <link href="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.98.0/css/materialize.min.css" rel="stylesheet" />
     <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css" />
+
+      <style>
+          .modal .modal-footer {
```
Comment (isheebo) [inline public/index.html]: could you move this to a separate css file

Comment (collinewait) [reply public/index.html]: It is resolved

File: src/containers/reportArticle.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	};
+
+	handleClick = () => {
+		$('#reportModal').modal('close');
+	};
+
+	render() {
+		return (
+			<ReportArticleForm
+				issue={this.state.issue}
```
Comment (isheebo) [inline src/containers/reportArticle.js]: please destructure the`state` object here

File: src/components/articles/reportArticleForm.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '../../styles/articleStyles/reportArticleForm.scss';
+
+const InputFields = (props) => {
+
+	const { value, id, handleChange, issue } = props;
```
Comment (isheebo) [inline src/components/articles/reportArticleForm.js]: change line 5  from
```js
const InputFields = (props) => {
```
to 
```js
const InputFields = ({ value, id, handleChange, issue }) => {
```
delete line 7

File: src/components/articles/reportArticleForm.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+	return (
+		<div>
```
Comment (isheebo) [inline src/components/articles/reportArticleForm.js]: The enclosing div happens to have no class, meaning it is just a container. I would prefer you use `React.Fragment` for this.

File: src/components/articles/reportArticleForm.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+				className='with-gap'
+				checked={issue === value}
+				onChange={handleChange}
+			/>
+			<label htmlFor={id} className='black-text'>{value}</label>
+		</div>
+	);
+};
+
+const ReportArticleForm = (props) => {
```
Comment (isheebo) [inline src/components/articles/reportArticleForm.js]: Repeat the procedure specified [here](https://github.com/andela/ah-frontend-iroquois/pull/40/files#r230080143) on line 24 and line 26

Comment (collinewait) [reply src/components/articles/reportArticleForm.js]: This has been resolved

File: src/components/articles/reportArticleForm.js (hunk 8 above)
Comment (collinewait) [reply src/components/articles/reportArticleForm.js]: The change has been made

File: src/components/articles/reportArticleForm.js (hunk 9 above)
Comment (collinewait) [reply src/components/articles/reportArticleForm.js]: I have replaced the `div` with `React.Fragment`

File: src/containers/reportArticle.js (hunk 7 above)
Comment (collinewait) [reply src/containers/reportArticle.js]: It has has been resolved.

**Review by reiosantos — DISMISSED** [review]

**Comment by reiosantos** [comment]
@collinewait, Please look at this error I am getting. It happens when the page loads

image

**Comment by collinewait** [comment]
> @collinewait, Please look at this error I am getting. It happens when the page loads
> 
> image

This has been worked on.

**Review by DrKimpatrick — DISMISSED** [review]

**Review by huxaiphaer — DISMISSED** [review]

**Review by reiosantos — DISMISSED** [review]

**Comment by isheebo** [comment]
Resolve conflicts

**Review by DrKimpatrick — APPROVED** [review]

**Review by reiosantos — APPROVED** [review]

**Pull request merged**